### ISSUE TITLE
Side drawer opening requires unpredictable touch zones

### ISSUE THREAD

**Issue by Victor239** [moffatman/chan#181]
When on an individual post:
- Swiping left-to-right from screen edge triggers the Android gesture navigation "back", which opens the "Exit the app?" prompt. Sometimes opens the side drawer too. 
- Swiping left-to-right from the middle of the screen goes back one screen, takes you to the imageboard home. Consistent. 
- Swiping right-to-left from screen edge triggers the Android gesture navigation "back", which opens the "Exit the app?" prompt. Consistent. 
- Unpredictable area to open the side drawer from this screen. Best chance of doing so seems to me to be swiping left-to-right using the bottom 1cm from the screen edge, but this also usually opens the app switcher. 

When on an imageboard home:
- Swiping left-to-right from screen edge triggers the Android gesture navigation "back", which opens the "Exit the app?" prompt. Sometimes opens the side drawer too. 
- Swiping left-to-right starting from the left half opens the side drawer. Consistent. 
- Swiping left-to-right starting from the right half opens the "Open boards" label. Consistent. 
- Swiping right-to-left from screen edge triggers the Android gesture navigation "back", which opens the "Exit the app?" prompt. Consistent. 

My ideal solution would be:
- Option to disable "Exit the app" prompts, instead open the side drawer when the swiping left-to-right from screen edge and triggering the Android gesture navigation "back".
- Don't open the side drawer AND start the Android "back" gesture simultaneously - choose one or the other. 

It's also unpredictable the touch zones of reply overlays, but I haven't done a full analysis of that yet.

**Comment by moffatman** [comment]
What phone / android version are you on? There is some API I still need to adopt for Android 14 predictive back. On my MI MIX 4,  back gesture is always used in lower ~2/3 of both edges, and drawer is pulled open in top ~1/3 of left edge. And this is the same as in Kuroba-EX so I assumed it was working properly. Not an Android guy. So thanks for the feedback!

One difference actually is that the back gesture in a thread will take me back to the catalog, not prompt to exit the app.

**Comment by Victor239** [comment]
Xiaomi Pocophone F1, Android 12 (LineageOS). Can't replicate the 1/3rd and 2/3rds behaviour you mentioned unfortunately but happy to test further.

**Comment by moffatman** [comment]
Could you do the same tests in some other app like kuroba? So we can isolate whether this is lineage specific maybe?

**Comment by Victor239** [comment]
Sure. What relevant settings do you use in KurobaEx under Behaviour?

**Comment by moffatman** [comment]
I think it's default. I didn't touch the gesture exclusion, and "allow screens to be swiped away", as well as "allow the thread screen to be swiped away" are enabled

**Comment by Victor239** [comment]
KurobaEx:

On an individual post:
- Any swipe will take me back to the imageboard home

On imageboard home:
- Left-to-right or right-to-left on screen edge will do the Android "back" gesture icon and will minimise the app to the background.
- Left-to-right anywhere not on screen edge will open the sidebar.

**Comment by KyleFN** [comment]
I'm also having this problem. It seems totally random whether swiping back will ask if I want to exit the app, go back to the all threads view or open the side drawer.

**Comment by moffatman** [comment]
Yes the back button handling has certain bugs, I've rewritten it completely now. So hopefully this gets better in the next update.

**Comment by Victor239** [comment]
This seems resolved for me, thanks!
